src/MDMTransition.h

```diff
@@ -34,6 +35,8 @@ NS_SWIFT_NAME(Transition)
  */
 - (void)startWithContext:(nonnull id<MDMTransitionContext>)context;
 
+@optional
+- (void)startWithInteractiveContext:(nonnull id<MDMInteractiveTransitionContext>)context;
```

Comment:
I generally prefer avoiding `@optional` these days because it means the compiler won't enforce method signatures.

Instead, I suggest we add an `MDMTransitionWithInteraction` protocol similar to how we've been building the other features.